ledger/participant-integration-api/src/main/scala/platform/store/backend/h2/H2StorageBackend.scala

```diff
@@ -197,7 +197,32 @@ private[backend] object H2StorageBackend
       connectionInitHook: Option[Connection => Unit],
   )(implicit loggingContext: LoggingContext): DataSource = {
     val h2DataSource = new org.h2.jdbcx.JdbcDataSource()
-    h2DataSource.setUrl(jdbcUrl)
+
+    // H2 (org.h2.jdbcx.JdbcDataSource) does not support setting the user/password within the jdbcUrl, so remove
+    // those properties from the url if present and set them separately. This helps run canton tests which set
+    // user/password. Note that Postgres and Oracle support user/password in the URLs, so we don't bother exposing
```

Comment:
> This helps run canton tests which set user/password.

Do we need to mention Canton here? The change makes sense even without Canton.

Reply:
Indeed it does, so I am moving the reference to canton to a parenthetical note the end (so the canton dependency is clear in the code, as no canton code is yet in the daml repo).